### ISSUE TITLE
[Discussion] Swipe Typing in Unexpected Keyboard

### ISSUE THREAD

**Issue by NeatNit** [Julow/Unexpected-Keyboard#484]
I like swipe typing. If you don't know what that is, it's this: https://youtu.be/SJ-RAefCG_c . This feature is available on most (maybe all) first-party virtual keyboards today - Samsung Keyboard, Gboard, I think even the default iOS VK.

I'm currently using Samsung Keyboard as my daily driver but would love to switch to a free-er alternative (there are also a bunch of annoying bugs with it). I absolutely love Unexpected Keyboard's feel! But obviously, it doesn't have swipe typing, meaning I'd be making a big sacrifice if I switch.

This is not exactly a request, because adding swipe typing is undeniably a huge undertaking. This is more of a question or discussion about *what if* Unexpected Keyboard had swipe typing. Could it be implemented without clashing with the core feature?

The obvious intuitive answer is no: UK and swipe typing define conflicting actions to the same input, swiping in some direction from a letter key. It's obvious that both features can't be accessible simultaneously. But I say this clash can be overcome!

I imagine it like this:

First of all, there would be an easily accessible key to enable or disable swipe typing. For example: swiping up from the keyboard (UK style) would toggle it, with a clear indication of the active state.

While ST is disabled, UK works exactly as it does now.

While enabled, swiping from *letter keys* will initiate a word swipe, ignoring the UK extra symbols per key. Tapping works normally, and symbol keys work the same as with ST disabled. If you want to access a letter key's extra symbols and want to skip the disable-type-enable hassle, you can tap-and-hold on a key for a short while and then swipe in the direction of the symbol. There should be visual and haptic feedback to make this all more clear.

So that's how I imagine it! In my head, this is the perfect virtual keyboard. What do you think? Does this clash with anything I haven't thought of? Would it clash with UK's design philosophy, or would it improve it?

Hypothetically (and I will stress *this is not likely to happen*), if I were to add this functionality and make a pull request, would it be rejected on merit or accepted subject to code quality and such?

**Comment by NeatNit** [comment]
Forgot this part:

There needs to be a corrections bar so you can choose between multiple candidates words, if the detected word isn't the one you meant.

I'm also completely neglecting the technical questions of how to take a swiped path and produce a list of candidate words, how to manage the dictionary, and all that noise. Here I just wanted to talk about the design and concepts.

**Comment by Julow** [comment]
Swipe typing is not necessarily incompatible with the current swipe gesture as the finger would travel a different distance and draw a different shape.

It requires an implementation of spell checking and word correction but this is already something that I'd like to have, independently of swipe typing.

I would not personally spend time on the implementation as I am busy with other features but I'm not against.

**Comment by NeatNit** [comment]
Alright, good to know! By the way, I think #475 was intended to be about this too.

**Comment by Emnolope** [comment]
Swiping would make the keyboard perfect.

**Comment by dilipkn007** [comment]
Swipe typing is the only reason why I switch back to gboard

**Comment by graphedge** [comment]
@dilipkn007 indeed,  Swipe typing is the only reason why I switch back to gboard.

**Comment by anaxonda** [comment]
Are the recently released [futo swiping models ](https://swipe.futo.tech/) useful here?

**Comment by kauesena** [comment]
Swipe typing would allow me to dispense with the Samsung Keyboard. Currently, I keep switching to Unexpected Keyboard when I need Ctrl or orther modifier key, but, since I am very slow at typing choosing each character at a time, I change it back to Samsung Keyboard to input text itself. Besides, swiping is more friendly to single-hand use of the phone.
